src/utils.js

```diff
@@ -87,13 +87,15 @@ export const int2hex = num => {
 /**
  * Converts a number to a big endian hexstring of a suitable size, optionally little endian
  * @param {number} num
- * @param {number} size - The required size in hex chars, eg 2 for Uint8, 4 for Uint16. Defaults to 2.
+ * @param {number} size - The required size in bytes, eg 1 for Uint8, 2 for Uint16. Defaults to 1.
  * @param {boolean} littleEndian - Encode the hex in little endian form
  * @return {string}
  */
-export const num2hexstring = (num, size = 2, littleEndian = false) => {
+export const num2hexstring = (num, size = 1, littleEndian = false) => {
   if (typeof num !== 'number') throw new Error('num must be numeric')
   if (num < 0) throw new RangeError('num is unsigned (>= 0)')
+  size = size * 2
+  if (size % 2 !== 0) throw new RangeError('size must be a multiple of 2')
```

Comment:
i think the error msg should be about it being an integer. If it is an integer, it will always be multiple of 2 as you multipled it by 2.

Reply:
Good point, it's essentially acting as an integer check there. I'll change that check/message

Reply:
mm the rest looks good, will merge once i see this change